### ISSUE TITLE
Android 11 Support

### ISSUE THREAD

**Issue by nsribn** [libhybris/libhybris#524]
****I am using running Libhybris in chroot on Android AOSP11. Running libhybris test is giving  below error .
Does libhybris  support Android 11 ( Seems to only q folder is support corresponding Android 10 and no r folder for Android11 whose linker sources has changed)**
I also applied Bionic patches ,android 11 kernel crashes** 

WARNING: linker Warning: "/system/apex/com.android.runtime/lib64/bionic/libdl.so" unused DT entry: unknown processor-specific (type 0x70000001 arg (nil)) (ignoring)
WARNING: linker Warning: "/system/lib64/ld-android.so" unused DT entry: unknown processor-specific (type 0x70000001 arg (nil)) (ignoring)
TLS relocations not yet implemented in libhybrisAborted`

**Comment by xuqiang918** [comment]
Do you resolved this it ?

**Comment by mlehtima** [comment]
Libhybris assumes there are certain patches applied to Android sources to handle TLS and other things.

**Comment by krnlyng** [comment]
>  I also applied Bionic patches ,android 11 kernel crashes**
> 

If you patch android, then the kernel won't boot android anymore. This is meant to run linux distributions on top of a patched kernel, not android.

**Comment by xuqiang918** [comment]
I get the same problem

**Comment by Edward-888** [comment]
do you resolved it？

**Comment by Edward-888** [comment]
> 

Do you resolved it？
I also add a folder r but I have the same problem；I found that the compiled library has extra tls symbols，I wonder if that's what's causing the problem？
Please let me know if you solve it ，thankyou~

**Comment by nsribn** [comment]
Yes, solved TLS issue. I applied bionic patches for android. Copied Bionic libraries to  chroot ubuntu running inside Android .
Tested Hybris linking against the new libraries path.
But unfortunately, I do not have HWC2 support to test glve2, my device support HIDL . So issue is not completely resolved

**Comment by Edward-888** [comment]
> 
where is the patches? I patched it from Halium's hybris-patches ,but it Still report this error. I saw Halium in the meantime and felt good.
